### PR TITLE
tests: Add memset() unit tests

### LINKED ISSUE #1859: Add unit test for memset()
Implement some test cases for standard memset()

### PR DESCRIPTION
**Pull request by KhaledEmaraDev**
Closes #1859

### CODE REVIEW

File: templates/x86/qemu/mods.config [hunk 1]
```diff
@@ -90,6 +90,7 @@ configuration conf {
 	@Runlevel(2) include embox.test.stdlib.qsort_test
 	@Runlevel(2) include embox.test.posix.environ_test
 	@Runlevel(2) include embox.test.posix.getopt_test
+	@Runlevel(2) include embox.test.stdlib.memset_test
```
Comment (anton-bondarev) [inline templates/x86/qemu/mods.config]: Rm this please

**Comment by anton-bondarev** [comment]
@KhaledEmaraDev Thanks.

but please format at first see code-style ([https://github.com/embox/embox/wiki/Code-Style](https://github.com/embox/embox/wiki/Code-Style) )

**Comment by KhaledEmaraDev** [comment]
@anton-bondarev Yeah, sorry about that. I somehow mixed up spaces and tabs, but didn't notice because it looked fine in my editor. I'll set up clang-format. Anyways, removed the test from mods.config and fixed up indentation.

**Comment by anton-bondarev** [comment]
@KhaledEmaraDev Please don't rebare your branch now, just push commits under branch head

File: src/tests/stdlib/memset.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	memset(str, -1, sizeof str);
+
+	test_assert_mem_equal(str, exp, sizeof str);
+}
+
+TEST_CASE("memset null") {
+	char str[] = {1, 2, 3, 4, 5};
+	char exp[] = {0, 0, 0, 0, 0};
+
+	memset(str, (int)NULL, sizeof str);
```
Comment (anton-bondarev) [inline src/tests/stdlib/memset.c]: Why do you use NULL, It's enough 0 in my mind

**Comment by anton-bondarev** [comment]
@KhaledEmaraDev Can you add more tests. for example
memset in the middle of buffer.
It checks that memset don't touch neighbour bytes

**Comment by KhaledEmaraDev** [comment]
@anton-bondarev Sorry about the delay. Corrected NULL to 0 and added tests for the middle of the buffer. Tests passed.

**Comment by anton-bondarev** [comment]
I've closed PR if you want to contribute these changes please open a new PR

**Comment by anton-bondarev** [comment]
I've closed PR if you want to contribute these changes please open a new PR

**Pull request closed without merging**